sources/api/migration/migrations/v1.11.0/aws-creds/src/main.rs

```diff
@@ -0,0 +1,20 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting for AWS credential configuration.
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&["settings.aws"]))
+}
```

Comment:
This needs to be scoped to the individual settings since we previously had `settings.aws.region` for many variants.

I don't recall whether `settings.aws` will be fully cleaned up if all the settings underneath are removed. If not, and if that causes a problem, we might need to conditionalize this based on variant. `pki-affected-services` is the only other time we've done that in a migration.

Reply:
> This needs to be scoped to the individual settings

Does this mean we need separate migrations for each property? So one each for `settings.aws.profile`, `settings.aws.config`, and `settings.aws.credentials`?

Reply:
We have a `AddSettingsMigration` migration helper that accept a list of settings: https://github.com/bottlerocket-os/bottlerocket/blob/aa3534b85c242919f148bd19f36d801790392b04/sources/api/migration/migration-helpers/src/common_migrations.rs#L6-L8

Example in use: https://github.com/bottlerocket-os/bottlerocket/blob/aa3534b85c242919f148bd19f36d801790392b04/sources/api/migration/migrations/archived/v1.0.8/kubelet-unsafe-sysctl-kube-reserved/src/main.rs#L9-L14